### ISSUE TITLE
Cypress 10.8.0: Much slower running in CI and randomly failing tests

### ISSUE THREAD

**Issue by vikingair** [cypress-io/cypress#23830]
### Current behavior

A successful test run with version 10.7.0: ~ 5 min
A successful test run with version 10.8.0: ~ 12 min

Also, with 10.7.0 we had our tests very stable failing almost never.

With Version 10.8.0 the tests started to run into several random timeouts, but this could be related to the very slow execution times.

Browser: Electron 102 (headless)

### Desired behavior

I would expected that Cypress would become faster with the recent update, because of the memory footprint optimizations.

### Test code to reproduce

// code should be not of relevance

### Cypress Version

10.8.0

### Node version

v16.14.0

### Operating System

Linux Ubuntu Latest GitHub Workflow

### Other

Possibly related: https://github.com/cypress-io/cypress/issues/23824

**Comment by ericyhwang** [comment]
Also seeing this issue - tests running much more slowly on Cypress 10.8.0, with some hanging while consuming 100% CPU for 2+ hours on CI, until manually stopped.

Reverting back to Cypress 10.7.0, the same test suite finishes in well under 10 minutes.

----

CI is using the Docker image `cypress/base:14.7.0` as a base, with Cypress 10.8.0 installed via NPM.

Exec-ing into a container with a hanging test, the main Cypress process, or at least the one with the lowest process id, was using 99% CPU, and the GPU process was using 37% CPU.

```
$ ps -elf
F S UID        PID  PPID  C PRI  NI ADDR SZ WCHAN  STIME TTY          TIME CMD
0 R root      1468    19 99  80   0 - 9942207 -    18:33 ?        00:10:35 /root/.cache/Cypress/10.8.0/Cypress/Cypress --no-sandbox -- --run-project ./ --ci-build-id Wed Sep 14 2022 18:33:52 GMT+0000 (Coordinated Universal Time) --env <redacted> --key <redacted> --output-path /tmp/tmp-19-QLkCqiVSY4TA --parallel --record true --spec ["<redacted>"] --tag <redacted> --cwd <redacted> --userNodePath /usr/local/bin/node --userNodeVersion 14.7.0
0 S root      1474  1468  0  80   0 - 8446627 SyS_pp 18:33 ?      00:00:00 /root/.cache/Cypress/10.8.0/Cypress/Cypress --type=zygote --no-zygote-sandbox --no-sandbox --enable-crashpad --enable-crashpad
0 S root      1475  1468  0  80   0 - 8446628 SyS_pp 18:33 ?      00:00:00 /root/.cache/Cypress/10.8.0/Cypress/Cypress --type=zygote --no-sandbox --enable-crashpad --enable-crashpad
1 S root      1627  1474 37  80   0 - 8488446 SyS_po 18:33 ?      00:03:03 /root/.cache/Cypress/10.8.0/Cypress/Cypress --type=gpu-process --no-sandbox --disable-dev-shm-usage --enable-crashpad --enable-crash-reporter=<redacted>,no_channel --user-data-dir=/root/.config/Cypress --gpu-preferences=WAAAAAAAAAAgAAAIAAAAAAAAAAAAAAAAAABgAAAAAAA4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAIAAAAAAAAAABAAAAAAAAAAgAAAAAAAAACAAAAAAAAAAIAAAAAAAAAA== --use-gl=angle --use-angle=swiftshader-webgl --shared-files --field-trial-handle=0,i,13177495466960604662,3729574163230753632,131072 --disable-features=SpareRendererForSitePerProcess
0 S root      1638  1468  2  80   0 - 679689 SyS_ep 18:33 ?       00:00:11 Cypress: Config Manager
0 S root      1651  1468  0  80   0 - 8462155 -    18:33 ?        00:00:00 /root/.cache/Cypress/10.8.0/Cypress/Cypress --type=utility --utility-sub-type=network.mojom.NetworkService --lang=en-US --service-sandbox-type=none --no-sandbox --disable-dev-shm-usage --ignore-certificate-errors=true --use-fake-device-for-media-stream --ignore-certificate-errors=true --enable-crashpad --enable-crash-reporter=<redacted>,no_channel --user-data-dir=/root/.config/Cypress --shared-files=v8_context_snapshot_data:100 --field-trial-handle=0,i,13177495466960604662,3729574163230753632,131072 --disable-features=SpareRendererForSitePerProcess --enable-crashpad
1 S root      1836  1475  8  80   0 - 14284109 -   18:34 ?        00:00:41 /root/.cache/Cypress/10.8.0/Cypress/Cypress --type=renderer --enable-crashpad --enable-crash-reporter=<redacted>,no_channel --user-data-dir=/root/.config/Cypress --app-path=/root/.cache/Cypress/10.8.0/Cypress/resources/app --enable-sandbox --no-sandbox --disable-dev-shm-usage --autoplay-policy=no-user-gesture-required --force-device-scale-factor=1 --use-fake-ui-for-media-stream --disable-gpu-compositing --lang=en-US --num-raster-threads=1 --renderer-client-id=4 --launch-time-ticks=6908206572 --shared-files=v8_context_snapshot_data:100 --field-trial-handle=0,i,13177495466960604662,3729574163230753632,131072 --disable-features=SpareRendererForSitePerProcess
0 S root      1849  1468  3  99  19 - 110216 -     18:34 ?        00:00:17 /root/.cache/Cypress/10.8.0/Cypress/resources/app/node_modules/@ffmpeg-installer/linux-x64/ffmpeg -f image2pipe -use_wallclock_as_timestamps 1 -i pipe:0 -y -vcodec libx264 -filter:v crop='floor(in_w/2)*2:floor(in_h/2)*2' -preset ultrafast <redacted>.mp4
```

**Comment by chrisbreiding** [comment]
That's definitely not good! I've been trying to reproduce this using our [kitchensink repo](https://github.com/cypress-io/cypress-example-kitchensink), but haven't had any luck. I haven't seen any performance differences between 10.7.0 and 10.8.0 in historical CI runs or when running locally, both inside and outside Docker. 

There's likely something in your tests or system setup that's being effected by changes in 10.8.0, but that isn't present in the kitchensink tests. 

Is there any more information you can provide about your tests? 

- How many spec files?
- How many total tests?
- Can you provide the contents of your `cypress.config` file?
- Can you run with debug logs (`DEBUG=cypress* cypress run`) and post the stdout from the run?

This is clearly a serious issue, but it will be difficult for us to pinpoint the cause of it without being able to reproduce it, so any information you provide can help. Thanks in advance!

**Comment by vikingair** [comment]
Sure.

- Number of spec files: 9
- Total tests: 12
- Cypress-Config (`cypress-audit` with version: 1.1.0)

```ts
import { defineConfig } from 'cypress';
import { readFileSync } from 'fs';
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import { lighthouse, pa11y, prepareAudit } from 'cypress-audit';

const conf = JSON.parse(readFileSync(process.env.CONF, 'utf-8'));

export default defineConfig({
    e2e: {
        setupNodeEvents(on, _config) {
            on('before:browser:launch', (_browser, launchOptions: any) => {
                prepareAudit(launchOptions);
            });

            on('task', {
                lighthouse: lighthouse(), // calling the function is important
                pa11y: pa11y(), // calling the function is important
            });
        },
        specPattern: 'src/**/*.cy.{js,jsx,ts,tsx}',
        baseUrl: conf.domain,
        env: {
            CONFIG: JSON.stringify(conf),
        },
        viewportWidth: 1200,
        viewportHeight: 800,
    },
});
```

> Can you run with debug logs (`DEBUG=cypress* cypress run`) and post the stdout from the run?

Will try to get this information for you asap. But it requires a little effort, because we had to rollback the update already.

Did you try to reproduce it with the `cypress-audit` integration in the kitchensik repo? Didn't test the run without it.

**Comment by vikingair** [comment]
I've tested it now from my local machine (MacOS).

Looking at the debug output, I cannot send you the output with good conscience, because it contains a lot of sensible information that I'd not like to share. (e.g. login credentials, JWTs and used URLs etc.)

But my observation was that my tests executed roughly with exactly the same speed on both Cypress versions. 

Our GitHub Actions workflow runs on "ubuntu-latest", but I currently have no linux users on my team to run it locally. If I get more insights, I'll keep you updated.

**Comment by mjhenkes** [comment]
@fdc-viktor-luft, could you attempt to re-create this in a docker container?

And is it correct that you're only seeing the slowdown inside a github Actions, not locally on your mac?

**Comment by vikingair** [comment]
@mjhenkes Yes, I can only reproduce this within GitHub Actions.

Didn't try it within a docker container yet. Will try to find some time to test it within a container. If I can reproduce it with our tests, I'll try to create a repro without all of our sensitive data for you.

**Comment by capc0** [comment]
We are experiencing the same issue within our GitLab CI Pipeline running in a docker container

`cypress/browsers:node14.17.0-chrome91-ff89`

```
  │ Cypress:        10.8.0                                                                         │
  │ Browser:        Electron 102 (headless)                                                        │
  │ Node Version:   v14.17.0 (/usr/local/bin/node)                                                 │
  │ Specs:          59 found
```

some errors that are printed within the log that might be related

```
[331:0929/204824.767614:ERROR:node_bindings.cc(276)] Most NODE_OPTIONs are not supported in packaged apps. See documentation for more details.
libva error: va_getDriverName() failed with unknown libva error,driver_name=(null)
[491:0929/204841.439660:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.
Couldn't determine Mocha version
```

our config

```ts
export default defineConfig({
    videosFolder: 'cypress/videos',
    screenshotsFolder: 'cypress/screenshots',
    downloadsFolder: 'cypress/downloads',
    fixturesFolder: 'cypress/fixtures',
    viewportWidth: 1920,
    viewportHeight: 1080,
    trashAssetsBeforeRuns: false,
    video: true,
    numTestsKeptInMemory: 0,
    reporter: 'cypress-multi-reporters',
    reporterOptions: {
        configFile: 'cypress/cypress-reporter-config.json',
    },
    e2e: {
        // We've imported your old cypress plugins here.
        // You may want to clean this up later by importing these.
        setupNodeEvents(on, config) {
            on('after:spec', async (spec, results) => {
                if (results && results.video) {
                    // Do we have failures for any retry attempts?
                    if (results?.tests?.every(test => test?.attempts?.every(attempt => attempt?.state !== 'failed'))) {
                        return rm(results.video, { recursive: true, });
                    }
                }
            });
            on('file:preprocessor', cypressTypeScriptPreprocessor);
            return registerCodeCoverageTasks(on, config);
        },
        baseUrl: 'http://localhost:4200',
        specPattern: './cypress/e2e/**/*',
        supportFile: './cypress/support/e2e.ts',
    },
});
```

**Comment by vikingair** [comment]
## Preparation

`Dockerfile`:

```
FROM node:16

RUN apt-get update
# see https://docs.cypress.io/guides/continuous-integration/introduction#Machine-requirements
RUN apt-get install -y libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb

RUN npm i -g pnpm

WORKDIR /usr/tests

COPY . .
RUN pnpm i
CMD pnpm test
```

`.dockerignore`:

```
node_modules
```

Having `cypress@10.7.0` installed and running:

```sh
docker build -f Dockerfile -t tmp-cypress-10.7.0 .
```

Same for `cypress@10.9.0`.

## Execution

### 10.7.0

```sh
docker run tmp-cypress-10.7.0
```

Runs into an error:

```
Error [ERR_LOADER_CHAIN_INCOMPLETE]: "file:///root/.cache/Cypress/10.7.0/Cypress/resources/app/node_modules/ts-node/esm/transpile-only.mjs 'resolve'" did not call the next hook in its chain and did not explicitly signal a short circuit. If this is intentional, include `shortCircuit: true` in the hook's return.
```

### 10.9.0

```sh
docker run tmp-cypress-10.9.0
```

The run times reflect very much the times I see on the CI. Also many failing tests due to timeouts. However, docker on MacOS is still not very fast in with some operations and I couldn't compare it due to the above error to the run times using 10.7.0. So these results might not be of a great value for you.

**Comment by mmonteiroc** [comment]
Same issue here with Azure Devops pipelines and angular project
In our case we migrated from 10.7.0 to 10.9.0 

And times passed from ~30 mins of agent running  to 1hour 30 mins

**Comment by mmonteiroc** [comment]
Do we have an update on this issue ? Is there any info that we could provide that would help you investigate ?

**Comment by AtofStryker** [comment]
@mmonteiroc would you be able to recreate in azure devops pipeline and invite us to the project to see the behavior? I can see if I can get a reproduction running in actions later today, but azure devops is a bit difficult for us to work with since it isn't exactly open source.

**Comment by mmonteiroc** [comment]
Not sure I can invite you ( as we are corporate, and thats IT department.... ) 
But I can run in our DevOps pipeline any command you want with any cypress version you want, and then provide you the raw output logs 
Would that be helpful @AtofStryker  ?

**Comment by AtofStryker** [comment]
@mmonteiroc That could be pretty difficult. Likely what I will do is take a stab at github actions with a similar docker approach mentioned above to see if I can get a reliable reproduction

**Comment by danielvianna** [comment]
@AtofStryker I have a similar issue, how can I invite you to a private project?

**Comment by Incarnation** [comment]
I am running cypress v10.10.0 upgraded from v10.6.0 and found out the runtime increase from 15 min to 30 min for my tests
using Buildkite pipeline and docker container
running node 16 
browser electron

**Comment by AtofStryker** [comment]
> @AtofStryker I have a similar issue, how can I invite you to a private project?

@danielvianna if its a github repo, you can invite my username to the organization/repo

**Comment by danielvianna** [comment]
@AtofStryker  It's a private gitlab repository using gitlab CI/CD, is that doable?

**Comment by AtofStryker** [comment]
@danielvianna that should work. My gitlab user profile is [here](https://gitlab.com/AtofStryker).

**Comment by danielvianna** [comment]
@AtofStryker - invited, let me know if you need higher permission levels, we also have MS teams that we can invite for faster collaboration

**Comment by AtofStryker** [comment]
@danielvianna Thank you. I saw the invite in Gitlab today. Today was fairly eventful, but hoping to take a look first thing Monday.

**Comment by AtofStryker** [comment]
@danielvianna had a bit of time to take a look through. It's a bit hard to see from the job itself since polling is at its limits, but did notice the acceptance tests are on a fairly old version of node. Have you tried bumping to 16+ and see if the issue improves at all? The CPU usage looks to be maxed out completely for a large majority of the tests. Have you tried throwing more compute resources at the job to try to get that number lower and see if performance improves? I wouldn't expect these things to be related to a minor version bump but figured it might be a start to rule some things out.

**Comment by vikingair** [comment]
In GitHub Actions we were running on Node > 16 running into these performance issues. And the used machine resources are pre-configured but very powerful 😥

**Comment by DamienCassou** [comment]
> I wouldn't expect these things to be related to a minor version bump but figured it might be a start to rule some things out.

It seems several persons have complained here at approximately the same time. Also, in our case, sticking to 10.7.0 fixes the issue. On the right, you will see the result of the PR reverting to 10.7.0:
![image](https://user-images.githubusercontent.com/217543/197553203-275c9245-e641-4c08-ab2f-f28d74c51ac6.png)

**Comment by AtofStryker** [comment]
> It seems several persons have complained here at approximately the same time. Also, in our case, sticking to 10.7.0 fixes the issue.

I want to be clear that there is some type of regression in 10.7.0 and don't want to invalidate what you are experiencing. What I am trying to rule out is resourcing and other external factors that may help isolate the problematic change. My guess is the memory footprint changes or some of the changes under the hood to webkit caused a side effect.

Triage team is meeting this afternoon. Since reproduction is limited, I have a few suggestions how we might be able to isolate the change.

**Comment by danielvianna** [comment]
Hi Atof, we will try the newest node version. But it also crashes on my computer as well . Basically crashes every 3/4 tests or long tests. All of that happened after we were forced to upgrade to newest versions after version 8 which was way faster and stable

**Comment by AtofStryker** [comment]
@danielvianna Interesting that it is happening locally as well. Does this happen in all browsers and what OS are you on?

**Comment by danielvianna** [comment]
It does on Electron and Chrome

MacOS Monterrey, Version 12.2.1 (21D62)
Mac mini (M1, 2020)
Memory 16GB

**Comment by Incarnation** [comment]
My issue is only on Electron
runtime increased from 15 min to 30 min after upgrade to v10.10.0 from v.10.6.0 running on Buildkite pipeline and docker container

after switch the browser to Chrome
the runtime drop from 30 min to 17 min

**Comment by danielvianna** [comment]
@AtofStryker . I'm not sure if you checked, but we bumped up the version of Node and tests became slower or didn't improve at all

**Comment by danielvianna** [comment]
I see @mschile is assigned for this task, I can invite you to our repository if you want

**Comment by mschile** [comment]
@danielvianna, yes, you can invite [me](https://gitlab.com/mschile) to your repository. I see you are upgrading from Cypress 8 to 10 so your issue may not be the same as described by the OP since they are upgrading from 10.7 to 10.8. If you run version 10.7 do you still see the performance issues?

Has anyone upgrading from 10.7 been able to create a reproduction that they can share?

**Comment by danielvianna** [comment]
@mschile invited

**Comment by panwenhua** [comment]
The same problem happens with version 10.11.0

10.7.0 doesn't have these issues

**Comment by liambutler** [comment]
And it's the same on Cypress 11.0.0. Reverting back to 10.7.0 led to videos that were still choppy, but were at least watchable

Edit- adding additional context because OP (@fdc-viktor-luft) expressed confusion. Another symptom of poor performance in can be seen in the video recordings. As well as tests running more slowly, you will see skipped frames/hanging in the video output. Whatever changes were made in 10.8.0 led to Cypress being more resource-hungry. Cypress was already struggling to run on GitHub Actions in 10.7.0, but 10.8.0 has made matters worse

I agree that this seems to be CPU related; reducing `numTestsKeptInMemory` to 0 yielded no performance improvement

**Comment by mschile** [comment]
@danielvianna, do you still see the same performance issues if you upgrade to Cypress 10.7? If so, could you log a separate issue since I don't want it to get conflated with users that are seeing performance issues going from 10.7 to 10.8+.

**Comment by danielvianna** [comment]
@mschile we had some tests stabilized since version 8.3.1
Then we jumped into the last 2/3 months to 9.7.0 then 10.10

We never tested the version 10.7, but we tried what AtofStryker said (upgrade Node) and it got slower
It looks like someone else is now responsible for the ticket? ryanthemanuel ?

We can show the issue via MS teams. I have one person full time just trying to fix these same tests for almost 2 months (not writing anything new)

**Comment by ryanthemanuel** [comment]
For those who have commented or liked this issue, I have created several test binary builds that are basically 10.8.0 with some pieces of functionality removed that were introduced between 10.7.0 and 10.8.0. For anyone that can try these out it would be helpful to track down where the problem was introduced:

win32: `npm install https://cdn.cypress.io/beta/npm/10.8.1/win32-x64/10.8-minus-webkit-3524235d9dee0aada9259c246d8933b03990d6d4/cypress.tgz`
darwin (x64): `npm install https://cdn.cypress.io/beta/npm/10.8.1/darwin-x64/10.8-minus-webkit-3524235d9dee0aada9259c246d8933b03990d6d4/cypress.tgz`
darwin (arm64): `npm install https://cdn.cypress.io/beta/npm/10.8.1/darwin-arm64/10.8-minus-webkit-3524235d9dee0aada9259c246d8933b03990d6d4/cypress.tgz`
linux (x64): `npm install https://cdn.cypress.io/beta/npm/10.8.1/linux-x64/10.8-minus-webkit-3524235d9dee0aada9259c246d8933b03990d6d4/cypress.tgz`

**Comment by mmonteiroc** [comment]
I will try to run it in our CI either today or monday. And will give you feedback regarding performance of it

**Comment by vikingair** [comment]
I ran now the "linux (x64)"* package in our CI. Works perfectly fine. Same performance as with "10.7.0"

\* `https://cdn.cypress.io/beta/npm/10.8.1/linux-x64/10.8-minus-webkit-3524235d9dee0aada9259c246d8933b03990d6d4/cypress.tgz`

**Comment by liambutler** [comment]
I've also run the Linux (x64) package, and can see that performance has improved. Video recordings actually show the entire run, rather than hanging after a few seconds

**Comment by stychu** [comment]
Same problem here. Upgrading from 10.4.0 to 11.0.1 results in performance decrease on CI. Usually tests run takes 10-12 min but after upgrade it takes 25 min+ or exceeds 30 min timeout.

Tried 
>darwin (x64): npm install https://cdn.cypress.io/beta/npm/10.8.1/darwin-x64/10.8-minus-webkit-3524235d9dee0aada9259c246d8933b03990d6d4/cypress.tgz 

Seems to not have an issue with performance degradation

**Comment by ryanthemanuel** [comment]
OK. The fact that the attached binary is working for people is super helpful in us triaging the problem.

One additional thing that would be helpful for folks to try is to turn video off in 10.7 and video off in 10.8 and compare those two. If the performance is similar, then likely the issue is due to some video changes we made between 10.7 and 10.8. Is there anyone able to test this out? @stychu, @liambutler, @fdc-viktor-luft?

**Comment by denis-domanskii** [comment]
Just migrated from 8.6.0 to 11.0.1: 
- components tests are 30% slower
- integration tests without changes 

Video recording is disabled. Tests were ran in CI, on aws c5d.large agents.

**Comment by vikingair** [comment]
So I used now in my Cypress-Config:
```js
export default defineConfig({
  "e2e": {
     // ... more stuff ...
    "video": false
  }
});
```

Running times:
- `10.7.0`: 4m 54s
- `10.8.0`: 4m 5s 👀 

It seems like Cypress became much faster without videos 😱

**Comment by liambutler** [comment]
Running times (all on GitHub Actions with video disabled):

`10.7.0`: 1m 31s
`10.8.0`: 1m 30s
`10.8.1 (experimental build)`: 1m 29s

So not too much of a difference. But my suite of tests is small right now.

I also ran them with video recording enabled:

`10.7.0`: 1m 45s
`10.8.0`: 1m 55s
`10.8.1 (experimental build)`: 1m 45s

**Comment by Michel73** [comment]
I have the same problem with cypress 10.11.0. On Windows, macOS and ubuntu (github actions). Furthermore, the videos made on GitHub actions hang on unbuntu (22.04).

**Comment by stychu** [comment]
@ryanthemanuel 
> OK. The fact that the attached binary is working for people is super helpful in us triaging the problem.
> 
> One additional thing that would be helpful for folks to try is to turn video off in 10.7 and video off in 10.8 and compare those two. If the performance is similar, then likely the issue is due to some video changes we made between 10.7 and 10.8. Is there anyone able to test this out? @stychu, @liambutler, @fdc-viktor-luft?

I tested with `11.0.1` and with `video: false`  and indeed it seems that is the issue here.

**Comment by ryanthemanuel** [comment]
I think I've narrowed it down to two potential commits:

https://github.com/cypress-io/cypress/commit/dc9e9dc62c7085b20cebf0392a0f680e2af98ef8

and 

https://github.com/cypress-io/cypress/commit/bcd7548d74fdb5dab4790f10f158bb4f3e9387bc

I created a binary with 10.8.0 with those two commits removed and looked at our cypress realworld app repo with 10.7.0, 10.8.0, and 10.8.0 minus those commits. 10.7.0 and 10.8.0 with those commits removed both run fine with the github action on that repo. 10.8.0 consistently fails on windows.

10.8.0 - https://github.com/cypress-io/cypress-realworld-app/actions/runs/3473366965
10.8.0 minus commits - https://github.com/cypress-io/cypress-realworld-app/actions/runs/3474252648/jobs/5808051919
10.7.0 - https://github.com/cypress-io/cypress-realworld-app/actions/runs/3473337554

With a reproduction available and a good starting point, I'm going to go ahead and route this to the e2e team for resolution

**Comment by ryanthemanuel** [comment]
@stychu, @liambutler, @fdc-viktor-luft and any others can you try this build to see if also improves your performance? It's similar to the last one, but is more targeted in what was removed to narrow things down even more:

`npm install https://cdn.cypress.io/beta/npm/10.8.1/linux-x64/10.8-minus-video-refactor-a8279f8104c46a046d37a71991aa0eccb329a202/cypress.tgz`

**Comment by vikingair** [comment]
@ryanthemanuel Tested it with the above version (`https://cdn.cypress.io/beta/npm/10.8.1/linux-x64/10.8-minus-video-refactor-a8279f8104c46a046d37a71991aa0eccb329a202/cypress.tgz`).

Running time was: 5m 22s

It is just a little slower than the `10.7.0` (about 15s, but this could also just be a problem in the pipeline). However, all tests passed reliably, which is one of the most important aspects for me.

**Comment by liambutler** [comment]
@ryanthemanuel also tested with `10.8-minus-video-refactor`

Running times were:
- Video disabled: 01:26
- Video enabled: 01:45

Each video recording also showed the spec in its entirety. They didn't freeze early on in the recording, as has been the case with recordings from GA 10.8.0 and above

**Comment by stychu** [comment]
@ryanthemanuel I've just checked `npm install https://cdn.cypress.io/beta/npm/10.8.1/linux-x64/10.8-minus-video-refactor-a8279f8104c46a046d37a71991aa0eccb329a202/cypress.tgz` and it seems it's running around the same as 10.4 which Im currently running daily.

**Comment by mjhenkes** [comment]
Hey all, I think we've identified the issue. Here is a temporary 10.8.0 binary with the fix applied. Could those who were seeing the issue try the binary out and let us know if it's fixed?

```
npm install https://cdn.cypress.io/beta/npm/10.8.1/linux-x64/matth/test/10.8.0-fix-df3484ba3a0e05d78f0781c144f470d50262283b/cypress.tgz
```

I think what was happening is that durning a refactor, for electron we stopped limiting videos frame capture to very 5th frame and instead were capturing every frame, which lead to smooth videos, but slow tests.

For the exact fix see: https://github.com/cypress-io/cypress/commit/d382f66ac35a73775f87ba10d390aaca03d80e98

I plan to get a pr to fix this out today for the next release of cypress.
